File: drools-spring-v2-app/src/main/java/drools/spring/example/facts/Doctor.java
package drools.spring.example.facts;

import javax.persistence.Entity;
import javax.persistence.PrimaryKeyJoinColumn;

@Entity
@PrimaryKeyJoinColumn(name = "id")
public class Doctor extends User{
	
	private String institution;
	
	public Doctor() {
		super();
	}

	public Doctor(Long id, String username, String password, String name, String surname, String institution) {
		super(id, username, password, name, surname, ROLE.DOCTOR);
		this.institution = institution;
	}

	public String getInstitution() {
		return institution;
	}

	public void setInstitution(String institution) {
		this.institution = institution;
	}
	
}

File: drools-spring-v2-app/src/main/frontend/src/app/model/doctor.ts
import { User } from '../model/user';
import { ROLE } from '../model/role.enum';

export class Doctor extends User{
    institution : string;
    constructor(username : string, password : string, name : string, surname : string, email : string, institution : string){
        super(username, password, name, surname, email, ROLE.DOCTOR);
        this.institution = institution;
    }
}
